File: Java/Ejercicio 2/GestionPedidos.java
import java.util.LinkedList;
import java.util.Queue;

public class GestionPedidos {
    private Queue<Pedido> pedidos = new LinkedList<>();

    public void agregarPedido(String cliente, String plato) {
        pedidos.add(new Pedido(cliente, plato));
        System.out.println("Pedido agregado exitosamente.");
    }

    public void mostrarPedidos() {
        System.out.println("\n--- Pedidos Pendientes ---");
        if (pedidos.isEmpty()) {
            System.out.println("No hay pedidos pendientes.");
        } else {
            int i = 1;
            for (Pedido pedido : pedidos) {
                System.out.println(i + ". " + pedido);
                i++;
            }
        }
    }

    public Pedido atenderPedido() {
        return pedidos.poll();
    }
}

File: Go/Ejercicio 2/gestion_pedidos.go
package main

import "fmt"

type GestionPedidos struct {
	Pedidos []Pedido
}

func NewGestionPedidos() *GestionPedidos {
	return &GestionPedidos{Pedidos: []Pedido{}}
}

func (gp *GestionPedidos) AgregarPedido(cliente, plato string) {
	gp.Pedidos = append(gp.Pedidos, Pedido{Cliente: cliente, Plato: plato})
	fmt.Println("Pedido agregado exitosamente.")
}

func (gp *GestionPedidos) MostrarPedidos() {
	fmt.Println("\n--- Pedidos Pendientes ---")
	if len(gp.Pedidos) == 0 {
		fmt.Println("No hay pedidos pendientes.")
	} else {
		for i, pedido := range gp.Pedidos {
			fmt.Printf("%d. %s\n", i+1, pedido)
		}
	}
}

func (gp *GestionPedidos) AtenderPedido() *Pedido {
	if len(gp.Pedidos) == 0 {
		return nil
	}
	pedido := gp.Pedidos[0]
	gp.Pedidos = gp.Pedidos[1:]
	return &pedido
}
